testing: mm test panic on github.

Seen in a random test run on a github pr close to master. May be something to watch for:
```
?   	decred.org/dcrdex/client/intl	[no test files]
2025-10-19 18:02:02.215 [INF] mm_TEST: Upgrading database from version 0 to 2
2025-10-19 18:02:02.533 [INF] mm_TEST: Upgrading database from version 0 to 2
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x21409ce]

goroutine 36 [running]:
decred.org/dcrdex/client/mm.(*boltEventLogDB).updateEvent.func1(0xc00053c540)
	/home/runner/work/dcrdex/dcrdex/client/mm/event_log.go:271 +0x8e
go.etcd.io/bbolt.(*DB).Update(0xc000662248, 0xc00008aec0)
	/home/runner/go/pkg/mod/go.etcd.io/bbolt@v1.3.11/db.go:886 +0xe5
decred.org/dcrdex/client/mm.(*boltEventLogDB).updateEvent(0xc0000617e0, 0x0)
	/home/runner/work/dcrdex/dcrdex/client/mm/event_log.go:269 +0x97
decred.org/dcrdex/client/mm.(*boltEventLogDB).listenForStoreEvents(0xc0000617e0, {0x2dd13c8, 0xc00062e8c0})
	/home/runner/work/dcrdex/dcrdex/client/mm/event_log.go:327 +0x54
decred.org/dcrdex/client/mm.newBoltEventLogDB.func2()
	/home/runner/work/dcrdex/dcrdex/client/mm/event_log.go:206 +0x50
created by decred.org/dcrdex/client/mm.newBoltEventLogDB in goroutine 35
	/home/runner/work/dcrdex/dcrdex/client/mm/event_log.go:205 +0x265
FAIL	decred.org/dcrdex/client/mm	0.547s
ok  	decred.org/dcrdex/client/mm/libxc	1.444s
```

--- a/client/mm/event_log.go
+++ b/client/mm/event_log.go
@@ -9,6 +9,7 @@ import (
 	"encoding/binary"
 	"encoding/json"
 	"fmt"
+	"sync"
 	"time"
 
 	"decred.org/dcrdex/client/asset"
@@ -176,7 +177,7 @@ var (
 	noPendingKey   = []byte("np")
 )
 
-func newBoltEventLogDB(ctx context.Context, path string, log dex.Logger) (*boltEventLogDB, error) {
+func newBoltEventLogDB(ctx context.Context, path string, wg *sync.WaitGroup, log dex.Logger) (*boltEventLogDB, error) {
 	db, err := bbolt.Open(path, 0600, nil)
 	if err != nil {
 		return nil, err
@@ -201,8 +202,9 @@ func newBoltEventLogDB(ctx context.Context, path string, log dex.Logger) (*boltE
 	if err != nil {
 		return nil, err
 	}
-
+	wg.Add(1)
 	go func() {
+		defer wg.Done()
 		eventLogDB.listenForStoreEvents(ctx)
 		db.Close()
 	}()
@@ -264,7 +266,8 @@ func calcFinalStateBasedOnEventDiff(runBucket, eventsBucket *bbolt.Bucket, event
 
 // updateEvent is called for each event that is popped off the updateEvent. If
 // the event already exists, it is updated. If it does not exist, it is added.
-// The stats for the run are also updated based on the event.
+// The stats for the run are also updated based on the event. update must not
+// be nil.
 func (db *boltEventLogDB) updateEvent(update *eventUpdate) {
 	if err := db.Update(func(tx *bbolt.Tx) error {
 		botRuns := tx.Bucket(botRunsBucket)
@@ -366,11 +369,6 @@ func parseRunKey(key []byte) (startTime int64, mkt *MarketWithHost, err error) {
 		Host:    string(pushes[3]),
 	}
 	return
-}
-
-func (db *boltEventLogDB) Close() error {
-	close(db.eventUpdates)
-	return db.DB.Close()
 }
 
 func (db *boltEventLogDB) storeCfgUpdate(runBucket *bbolt.Bucket, newCfg *BotConfig, timestamp int64) error {

--- a/client/mm/event_log_test.go
+++ b/client/mm/event_log_test.go
@@ -10,6 +10,7 @@ import (
 	"maps"
 	"path/filepath"
 	"reflect"
+	"sync"
 	"testing"
 	"time"
 
@@ -36,7 +37,8 @@ func TestEventLogDB(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	db, err := newBoltEventLogDB(ctx, filepath.Join(dir, "event_log.db"), tLogger)
+	var wg sync.WaitGroup
+	db, err := newBoltEventLogDB(ctx, filepath.Join(dir, "event_log.db"), &wg, tLogger)
 	if err != nil {
 		t.Fatalf("error creating event log db: %v", err)
 	}
@@ -457,7 +459,8 @@ func TestUpdateFinalBalanceDueToEventDiff(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	db, err := newBoltEventLogDB(ctx, filepath.Join(dir, "event_log.db"), tLogger)
+	var wg sync.WaitGroup
+	db, err := newBoltEventLogDB(ctx, filepath.Join(dir, "event_log.db"), &wg, tLogger)
 	if err != nil {
 		t.Fatalf("error creating event log db: %v", err)
 	}

--- a/client/mm/event_log_upgrades_test.go
+++ b/client/mm/event_log_upgrades_test.go
@@ -6,6 +6,7 @@ import (
 	"encoding/json"
 	"fmt"
 	"path/filepath"
+	"sync"
 	"testing"
 	"time"
 
@@ -42,7 +43,8 @@ func TestEventLogV2Upgrade(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	db2, err := newBoltEventLogDB(ctx, dbPath, dex.StdOutLogger("TEST", dex.LevelError))
+	var wg sync.WaitGroup
+	db2, err := newBoltEventLogDB(ctx, dbPath, &wg, dex.StdOutLogger("TEST", dex.LevelError))
 	if err != nil {
 		t.Fatalf("Failed to create DB for upgrade: %v", err)
 	}
@@ -56,8 +58,6 @@ func TestEventLogV2Upgrade(t *testing.T) {
 		t.Fatalf("Expected version 2 after upgrade, got %d", version)
 	}
 	verifyV2EventLogUpgrade(t, db2.DB)
-
-	db2.Close()
 }
 
 func setupV1EventLogData(tx *bbolt.Tx) error {

--- a/client/mm/mm.go
+++ b/client/mm/mm.go
@@ -685,15 +685,14 @@ func (m *MarketMaker) Connect(ctx context.Context) (*sync.WaitGroup, error) {
 		}
 	}
 
-	eventLogDB, err := newBoltEventLogDB(ctx, m.eventLogDBPath, m.log.SubLogger("eventlogdb"))
+	var wg sync.WaitGroup
+	eventLogDB, err := newBoltEventLogDB(ctx, m.eventLogDBPath, &wg, m.log.SubLogger("eventlogdb"))
 	if err != nil {
 		return nil, fmt.Errorf("error creating event log DB: %v", err)
 	}
 	m.eventLogDB = eventLogDB
 
 	m.oracle = newPriceOracle(m.ctx, m.log.SubLogger("oracle"))
-
-	var wg sync.WaitGroup
 
 	wg.Add(1)
 	go func() {

